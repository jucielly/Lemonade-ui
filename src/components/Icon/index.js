import { ReactComponent as ArrowRight } from "./arrowRight.svg";
import { ReactComponent as Check } from "./check.svg";
import { ReactComponent as Refresh } from "./refresh.svg";
import PropTypes from "prop-types";
import React from "react";

export const icons = {
  "arrow-right": ArrowRight,
  check: Check,
  refresh: Refresh,
};

const Icon = ({ icon }) => {
  const IconComponent = icons[icon] || React.Fragment;
  return <IconComponent data-testid={`${icon}-icon`} />;
};

Icon.propTypes = {
  icon: PropTypes.oneOf(Object.keys(icons)),
};

export default Icon;
