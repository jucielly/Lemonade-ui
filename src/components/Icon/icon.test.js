import { render, screen } from "@testing-library/react";
import Icon, { icons } from ".";

describe("icon", () => {
  Object.keys(icons).forEach((icon) =>
    it(`Should match snapshot for ${icon}`, () => {
      const { baseElement } = render(<Icon icon={icon} />);

      expect(baseElement).toMatchSnapshot();
    })
  );
});
