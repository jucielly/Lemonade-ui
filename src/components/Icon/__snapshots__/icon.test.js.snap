// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon Should match snapshot for arrow-right 1`] = `
<body>
  <div>
    <svg
      data-testid="arrow-right-icon"
    >
      arrowRight.svg
    </svg>
  </div>
</body>
`;

exports[`icon Should match snapshot for check 1`] = `
<body>
  <div>
    <svg
      data-testid="check-icon"
    >
      check.svg
    </svg>
  </div>
</body>
`;

exports[`icon Should match snapshot for refresh 1`] = `
<body>
  <div>
    <svg
      data-testid="refresh-icon"
    >
      refresh.svg
    </svg>
  </div>
</body>
`;
